### ISSUE TITLE
[Modern] Is relay-compiler --watch supposed to work?

### ISSUE THREAD

**Issue by abergenw** [facebook/relay#1654]
```
$ relay-compiler --src ./src --schema schema.graphql --watch
TypeError: Cannot read property 'some' of undefined
    at hasWatchmanRootFile (/Users/andreas/Work/code/relay-modern-hello-world/node_modules/relay-compiler/bin/relay-compiler:208:29)
    at /Users/andreas/Work/code/relay-modern-hello-world/node_modules/relay-compiler/bin/relay-compiler:90:28
    at next (native)
    at step (/Users/andreas/Work/code/relay-modern-hello-world/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
    at /Users/andreas/Work/code/relay-modern-hello-world/node_modules/babel-runtime/helpers/asyncToGenerator.js:35:14
    at Promise.F (/Users/andreas/Work/code/relay-modern-hello-world/node_modules/core-js/library/modules/_export.js:35:28)
    at /Users/andreas/Work/code/relay-modern-hello-world/node_modules/babel-runtime/helpers/asyncToGenerator.js:14:12
    at run (/Users/andreas/Work/code/relay-modern-hello-world/node_modules/relay-compiler/bin/relay-compiler:126:18)
    at Object.<anonymous> (/Users/andreas/Work/code/relay-modern-hello-world/node_modules/relay-compiler/bin/relay-compiler:164:2)
    at __webpack_require__ (/Users/andreas/Work/code/relay-modern-hello-world/node_modules/relay-compiler/bin/relay-compiler:33:30)
```

**Comment by YasserKaddour** [comment]
I think you need to install watchman on windows and mac.

See #1644

**Comment by abergenw** [comment]
@YasserKaddour that's not the problem, watchman is installed and working... Without watchman the compiler doesn't get this far.

**Comment by eugene1g** [comment]
Same problem here - looks like something gets messed up during the compilation of the sourcecode.

**Comment by AndrewIngram** [comment]
Same here, i've only been able to use it without --watch

**Comment by yantonsoup** [comment]
Same problem here as well

**Comment by alangenfeld** [comment]
fix on the way

**Comment by leebyron** [comment]
Fixed in b730ba3812124b7d69e94ceb59d35e1f94cdf51b. We'll have this out in the next RC

**Comment by amiuhle** [comment]
When will this be published to npm?
